modules/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3ClientSettingsTests.java

```diff
@@ -167,14 +170,14 @@ public void testUseChunkedEncodingCanBeSet() {
         assertThat(settings.get("other").disableChunkedEncoding, is(true));
     }
 
-    public void testRegionCanBeSet() {
+    public void testRegionCanBeSet() throws IOException {
         final String region = randomAlphaOfLength(5);
         final Map<String, S3ClientSettings> settings = S3ClientSettings.load(
             Settings.builder().put("s3.client.other.region", region).build()
         );
         assertThat(settings.get("default").region, is(""));
         assertThat(settings.get("other").region, is(region));
-        try (S3Service s3Service = new S3Service()) {
+        try (S3Service s3Service = new S3Service(Mockito.mock(Environment.class))) {
```

Comment:
nit: we have `TestEnvironment.newEnvironment(Settings)` if you don't really need to mock class behavior.